samples/hello_world/README.rst

```diff
@@ -35,6 +36,24 @@ single thread: :file:`prj_single.conf`:
    :goals: run
    :compact:
 
+To build the semihosting version (ARM boards only), use the supplied
```

Comment:
The only other reference we have to semihosting in the Zephyr documentation talks about disabling it (in https://docs.zephyrproject.org/latest/application/application.html?highlight=semihosting). We could use a (brief) reference to what semihosting is somewhere (maybe a new section in this same application development doc) that we can reference in this sample.  Otherwise, the only documentation about semihosting will be buried here in hello_world.